backend/app/api/v1/endpoints/auth.py

```diff
@@ -0,0 +1,214 @@
+from typing import Annotated
+
+from fastapi import (
+    APIRouter,
+    Cookie,
+    Depends,
+    File,
+    Form,
+    HTTPException,
+    Request,
+    UploadFile,
+    status,
+)
+from fastapi.responses import JSONResponse
+from fastapi.security import OAuth2PasswordRequestForm
+from pydantic import ValidationError
+
+from app.core.constants import (
+    EMAIL_INVALID_MESSAGE,
+    LOGGED_IN_MESSAGE,
+    PASSWORD_RESET_MESSAGE,
+    TOKEN_REFRESH_MESSAGE,
+)
+from app.core.dependencies import get_current_user
+from app.interface.user_registration_interface import UserRegistrationInterface
+from app.schema.auth_schema import (
+    CreateUserRequest,
+    CreateUserResponse,
+    LoginResponse,
+    ProfileUpdateRequest,
+    UserProfileResponse,
+)
+from app.schema.response_schema import SuccessResponse
+from app.services.auth_service import AuthInterface, AuthService
+from app.services.image_service import image_service
+from app.services.jwt_token_service import JWTTokenInterface, JWTTokenService
+from app.services.user_registration_service import UserRegistrationService
+
+router = APIRouter(prefix="/auth", tags=["Authentication"])
+
+
+@router.post(
+    "/register", response_model=CreateUserResponse, status_code=status.HTTP_200_OK
+)
+def register(
+    full_name: str = Form(...),
+    email: str = Form(...),
+    password: str = Form(..., min_length=6),
+    avatar: UploadFile = File(default=None),
+    auth_service: AuthInterface = Depends(AuthService),
+):
+    avatar_path = None
+    if avatar:
+        avatar_path = image_service.save_image(avatar)
+    try:
+        create_user_request = CreateUserRequest(
+            full_name=full_name, email=email, password=password, avatar=avatar_path
+        )
+    except ValidationError:
```

Comment:
@ami-wtag bhai, Previously, this ValidationError was caught by the base exception handler. Since my base exception handler was not robust enough, it resulted in an internal server error in api endpoint response. To prevent this, I am now handling this exception specifically within this context. In my case it's handling this specific error when email is invalid.